### PR TITLE
fix: Ensure size estimate is applied to text shapes while typing

### PR DESCRIPTION
**Pull request by KDSBrowne**
### What does this PR do?
This PR introduces calculations to estimate and apply the size of text shapes in `tldraw-v1` as users type. The implementation leverages a utility function, TextUtil, that calculates the dimensions of the text based on its content and style properties. This approach mirrors the functionality in tldraw v1, where similar calculations are handled internally. The key improvement is ensuring that text shapes always have a defined size, preventing issues where size attributes might be missing during the editing process.

### Motivation

The motivation for this PR is to address the issue where `tldraw-v1` did not dynamically apply a size to text shapes while typing. Previously, tldraw-v1 calculated the size of text shapes during the `onCommand` event, which occurs after typing is completed. This approach could lead to inconsistencies, as the size was not updated in real-time. This PR ensures that text shapes have a defined size throughout the editing process.

Related to #20690

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/whiteboard/utils.js [hunk 1]
```diff
@@ -275,11 +275,109 @@ const getTextSize = (text, style, padding) => {
   return [width + padding, height + padding];
 };
 
+const TextUtil = {
+  getBoundsEstimate(shape) {
+    const { text, style } = shape;
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/whiteboard/utils.js:280]: Could you please add a comment to show where this is from? For example:

```
// getBoundsEstimate here is a copy of a function of the same name from <link to somewhere in github tldraw v1>
// ...

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {
 import Styled from './styles';
 import PanToolInjector from './pan-tool-injector/component';
 import {
-  findRemoved, filterInvalidShapes, mapLanguage, sendShapeChanges, usePrevious,
+  findRemoved, filterInvalidShapes, mapLanguage, sendShapeChanges, usePrevious, TextUtil
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx:19]:
```suggestion
  findRemoved, filterInvalidShapes, mapLanguage, sendShapeChanges, usePrevious, TextUtil,

```

**Comment by antobinary** [comment]
And just to be clear on the relation to #20690:
Here we ensure each text shape contains `size`. Therefore when it's time for exporting, we don't hit the edge case that we show a warning for in https://github.com/bigbluebutton/bigbluebutton/pull/20786

So to test this, it seems we need a bunch of typing to happen, exporting, and never to see the warning from 20786, right?

**Review by antonbsa — APPROVED** [review]
